politeiawww/cmd/politeiawwwcli/commands/subscribe.go

```diff
@@ -0,0 +1,112 @@
+package commands
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http/cookiejar"
+	"net/url"
+	"os"
+
+	v1 "github.com/decred/politeia/politeiawww/api/v1"
+	"github.com/decred/politeia/util"
+	"github.com/gorilla/websocket"
+	"golang.org/x/net/publicsuffix"
+)
+
+var SubscribeCmdHelpMsg = `subscribe [auth] <ping...>
+
+Connect and subcribe to www websocket. If auth is provided the connection will
+be made to the authenticated websocket (must be logged in).
+
+Supported commands:
+	- ping (does not require authentication)
+
+Request:
+{
+}
+`
+
+type Subscribe struct{}
+
+func (cmd *Subscribe) Execute(args []string) error {
+	// Check for user identity
+	if cfg.Identity == nil {
```

Comment:
Do you need to check for a user identity here?  You're not signing anything and unauthenticated connections are allowed so an identity should not be required.

Reply:
This is to make dead sure we are not in some sort of half logged in/off state. This makes it explicit and fails hard. This should also never happen but my OCD dictated it to be checked and handled.